### PR TITLE
feat: support codespaces

### LINKED ISSUE #1255: Support GitHub Codespaces
**Is your feature request related to a problem? Please describe.**
Let open-source contributors easily open the project in [GitHub Codespaces](https://github.com/features/codespaces).
It is a cloud-based IDE to let the contributors open with one click.
With some config files in the repo, the IDE should have everything ready at start. For example, the tests should be ready to run.

It would make contributing to the repo much easier.

**Describe the solution you'd like**
Here is an example setting. [[Link](https://github.com/keras-team/autokeras/tree/master/.devcontainer)]
Another example: [[Link](https://github.com/keras-team/keras/tree/master/.devcontainer)]

### PR DESCRIPTION
**Pull request by nautaa**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
close #1255
init codespaces develop environment.

codespaces docs: https://docs.github.com/en/codespaces

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM 4pdosc/hybridsql:latest
```
Comment (aceforeverd) [inline .devcontainer/Dockerfile:1]: centos 7 is a bit old.
I think we can just use ubuntu:20.04.

**Review by aceforeverd — APPROVED** [review]

**Review by zhanghaohit — APPROVED** [review]

**Comment by vagetablechicken** [comment]
so won't use ubuntu:20.04? plz make a comment @nautaa

**Comment by nautaa** [comment]
> so won't use ubuntu:20.04? plz make a comment @nautaa

yep, from my test, ubuntu is OK, but I think the development environment should be consistent with our docker image and cicd environment to facilitate the reproduction of some problems? And I am not familiar with the development environment of java, some configuration of the java environment may be required. Maybe someone will improve codespaces environment in the future.

**Comment by vagetablechicken** [comment]
OK, we can improve it next time.

**Pull request merged**